### ISSUE TITLE
quantization using openvino cannot handle when target in dataloader is not a tensor

### ISSUE THREAD

**Issue by yangw1234** [intel-analytics/BigDL#5229]
The following code will fail without adding `target_transform=transforms.Lambda(lambda label: torch.tensor(label, dtype=torch.long))`

```python

import torch
from torchvision import transforms
from torchvision.datasets import OxfordIIITPet
from torch.utils.data.dataloader import DataLoader
import torch
from torchvision.models import resnet18
from bigdl.nano.pytorch import Trainer
from torchmetrics import Accuracy


def finetune_pet_dataset(model_ft):

    train_transform = transforms.Compose([transforms.Resize(256),
                                          transforms.RandomCrop(224),
                                          transforms.RandomHorizontalFlip(),
                                          transforms.ColorJitter(brightness=.5, hue=.3),
                                          transforms.ToTensor(),
                                          transforms.Normalize([0.485, 0.456, 0.406],
                                                               [0.229, 0.224, 0.225])])
    val_transform = transforms.Compose([transforms.Resize([224, 224]),
                                        transforms.ToTensor(),
                                        transforms.Normalize([0.485, 0.456, 0.406],
                                                             [0.229, 0.224, 0.225])])

    # Apply data augmentation to the tarin_dataset
    train_dataset = OxfordIIITPet(root="/tmp/data",
                                  transform=train_transform,
                                  # target_transform=transforms.Lambda(
                                  #    lambda label: torch.tensor(label, dtype=torch.long)),
                                  download=True)
    val_dataset = OxfordIIITPet(root="/tmp/data",
                                transform=val_transform,
                                # target_transform=transforms.Lambda(
                                #     lambda label: torch.tensor(label, dtype=torch.long)),
                                )

    # obtain training indices that will be used for validation
    indices = torch.randperm(len(train_dataset))
    val_size = len(train_dataset) // 4
    train_dataset = torch.utils.data.Subset(train_dataset, indices[:-val_size])
    val_dataset = torch.utils.data.Subset(val_dataset, indices[-val_size:])

    # prepare data loaders
    train_dataloader = DataLoader(train_dataset, batch_size=32)
    val_dataloader = DataLoader(val_dataset, batch_size=32)

    num_ftrs = model_ft.fc.in_features

    # Here the size of each output sample is set to 37.
    model_ft.fc = torch.nn.Linear(num_ftrs, 37)
    loss_ft = torch.nn.CrossEntropyLoss()
    optimizer_ft = torch.optim.SGD(model_ft.parameters(), lr=0.01, momentum=0.9, weight_decay=5e-4)

    # Compile our model with loss function, optimizer.
    model = Trainer.compile(model_ft, loss_ft, optimizer_ft, metrics=[Accuracy()])
    trainer = Trainer(max_epochs=1)
    trainer.fit(model, train_dataloaders=train_dataloader, val_dataloaders=val_dataloader)

    return model, train_dataset, val_dataset


if __name__ == "__main__":

    model = resnet18(pretrained=True)

    _, train_dataset, val_dataset = finetune_pet_dataset(model)

    for data in val_dataset:
        print(data)
        break

    # Sample Inference Data
    x = torch.stack([val_dataset[0][0], val_dataset[1][0]])

    # Normal Inference
    model.eval()
    y_hat = model(x)
    predictions = y_hat.argmax(dim=1)
    print(predictions)

    # Static Quantization for OpenVINO
    q_model = Trainer.quantize(model,
                               accelerator='openvino',
                               calib_dataloader=DataLoader(train_dataset, batch_size=32))

    # Inference with Quantized Model
    y_hat = q_model(x)
    predictions = y_hat.argmax(dim=1)
    print(predictions)
```

**Comment by zhentaocc** [comment]
Do you have error msg? I can't download your dataset to reproduce your issue from my side.

**Comment by zhentaocc** [comment]
Can you provide the format of dataloader? I think it should be a tensor.

**Comment by yangw1234** [comment]
> Can you provide the format of dataloader? I think it should be a tensor.

the target is just an `int`

**Comment by zhentaocc** [comment]
You can close this if changes work.
